python-service/src/health_aggregator.py

```
@@ -0,0 +1,97 @@
+        return [{"data": json.loads(r[0]), "ts": r[1]} for r in rows]
+
+    def purge_old_snapshots(self, service: str, before_ts: int) -> None:
+        conn = sqlite3.connect(self.db_path)
+        conn.execute(
+            f"DELETE FROM health_snapshots WHERE service = '{service}' AND ts < {before_ts}"
+        )
+        conn.commit()
+        conn.close()
+
+    def run_diagnostic(self, host: str) -> str:
+        result = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)